asabdiscovery/proxy/handler.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+import aiohttp
+import aiohttp.web
+import asab.web.auth
+
+import asab.web.rest
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+
+class ProxyWebHandler:
+
+	def __init__(self, app, discovery_service):
+		self.App = app
+		self.DiscoveryService = discovery_service
+
+		web_app = app.WebContainer.WebApp
+
+		# Add the proxy endpoint(s)
+		# Example: /~/service_id/<service_id>/<tenant>/<object>/<method>
+		web_app.router.add_route(
+			'*',  # Accept all HTTP methods (GET, POST, etc.)
+			r"/~/{key}/{value}/{proxy_path:.*}",
+			self.proxy_by_key,
+		)
+
+
+	async def proxy_by_key(self, request):
+		key = request.match_info["key"]
+		value = request.match_info["value"]
+		proxy_path = request.match_info["proxy_path"]
+
+		# Locate URLs using the key-value pair
+		# TODO: Is this safe?
+		urls = await self.DiscoveryService.locate(**{key: value})
```

Comment:
Well. This is not very safe ... there should be at least some kind of validation of the key.